runtime/vm/BytecodeInterpreter.hpp

```diff
@@ -5228,6 +5228,7 @@ class INTERPRETER_CLASS
 					restoreInternalNativeStackFrame(REGISTER_ARGS);
 					/* Handle the virtual thread Object.wait call. */
 					J9VMJAVALANGVIRTUALTHREAD_SET_NOTIFIED(_currentThread, _currentThread->threadObject, JNI_FALSE);
+					J9VMJAVALANGVIRTUALTHREAD_SET_TIMEOUT(_currentThread, _currentThread->threadObject, millis);
```

Comment:
`(_currentThread, _currentThread->threadObject, millis)` should be something like `(_currentThread, _currentThread->threadObject, millis + (nanos*1000000))